src/ast/passes/callback_visitor.h

```diff
@@ -2,50 +2,23 @@
 
 #include <functional>
 
-#include "ast/visitors.h"
+#include "ast/visitor.h"
 
 namespace bpftrace {
 namespace ast {
 
 using callback = std::function<void(Node *)>;
 
-class CallbackVisitor : public Visitor {
+class CallbackVisitor : public Visitor<CallbackVisitor> {
 public:
   explicit CallbackVisitor(ASTContext &ctx, callback func)
-      : Visitor(ctx), func_(func)
+      : Visitor<CallbackVisitor>(ctx), func_(func)
   {
   }
-
-  void visit(Integer &integer) override;
```

Comment:
Seems like we're going to lose `override` catching us on typos. Any known tricks we can apply here for some more guardrails? Regardless, probably fine in exchange for the performance and simplicity win.

Reply:
I believe that the compiler should still be able to resolve the functions statically (even if marked virtual), and I actually started with that. I ultimately decided that the likelihood of typos wasn't worth the extra characters (& potential confusion from readers), since it's just "visit" and you're likely to declare in one block up front. WDYT?

Reply:
Yeah, I suppose only thing you could typo is "visit". Anything that manipulates AST is unit testable, so risk is low. Let's keep it as is, then.